### ISSUE TITLE
Repeatable rows repeat link labels

### ISSUE THREAD

**Issue by robincornett** [easydigitaldownloads/easy-digital-downloads#7702]
### Expected behavior
Links which include additional screen reader text should not repeat the text.

### Actual behavior
The "Remove" links in mextabox.php (https://github.com/easydigitaldownloads/easy-digital-downloads/blob/master/includes/admin/downloads/metabox.php#L443) properly add screen reader text to indicate what item is being removed, but both the visual link and the hidden text include the word "Remove", so it's read twice for screen reader users. Here's the output to clarify:

```html
<a class="edd-remove-row edd-delete" data-type="price">Remove<span class="screen-reader-text">Remove price option 1</span></a>
```

### Steps to reproduce the behavior
1) Inspect the "Remove" link for any repeatable row
2) If you don't have a screen reader handy, use dev tools to inspect the accessibility properties of the link. Firefox shows:
![image](https://user-images.githubusercontent.com/4853066/78259139-464eb300-74ca-11ea-9930-d49a5963e41e.png)

### Information (if a specific version is affected):
PHP Version: 7.3.5
EDD Version (or branch): master, 3.0
WordPress Version: 5.4

Any other relevant information:
This will also apply to any addons which implement repeater rows in their own metaboxes and use core EDD as an example.

The correct output would be:
```html
<a class="edd-remove-row edd-delete" data-type="price">Remove<span class="screen-reader-text"> price option 1</span></a>
```
which would keep the visual appearance consistent, and remove the repetition.

**Comment by spencerfinnell** [comment]
@robincornett What about using an `aria-label` for the current full string that screen readers are using now, which will in turn make it skip the inner text?

```php
/* translators: %s Download price option */ 
$aria_label = esc_attr_e( 'Remove price option "%s"', 'easy-digital-downloads );

echo '<a aria-label="' . esc_attr( $aria_label ) . '">' . esc_html_x( 'Remove', 'remove a download price option', 'easy-digital-downloads ) . '</a>'
```

**Comment by robincornett** [comment]
I think the current markup is semantically correct and sufficient, so the aria-label feels like an unneeded layer of complexity. Also, I think the `screen-reader-text` is a more reliable approach:

> While you could certainly do this, the main upside to using screen reader only text over an aria-label is that the screen reader is not required to support ARIA, it’s simply reading text on the page that sighted users cannot see. While most screen readers support ARIA, you may not want to rely on it if you don’t have to. ([source](https://accessible360.com/accessible360-blog/use-aria-label-screen-reader-text/))

**Comment by robincornett** [comment]
Technically, I think these should also be converted to button elements as they're not actually links.

**Comment by spencerfinnell** [comment]
@robincornett +1 on the `<button>` for sure.

My concern with:

```
<button>Remove <span class="screen-reader-text">Price Option 1</span></button>
```

is translating the `Remove` verb separately from the noun that will be read as a sentence. It forces it to  always be first, which is not the case for all languages.

We would either need to include the HTML inside the translation or add placeholders for the HTML as well.

Core seems to be using similar `aria-label` techniques to create more rich labels for similar actions:

https://github.com/WordPress/WordPress/blob/4f151b577e64f94395eac7d28d0f9ae28c66d58b/wp-admin/includes/class-wp-media-list-table.php#L389-L394
https://github.com/WordPress/WordPress/blob/4803fc405e3448e825bbc63b23cde2a6a86225a9/wp-includes/customize/class-wp-customize-nav-menu-location-control.php#L85-L86

**Comment by robincornett** [comment]
Ah, I didn't consider word order in translations, but that totally makes sense. My general preference is definitely not to include HTML markup in translatable strings (this may be just me), so I'll see about revising this. Thanks!

I'll make a separate issue about the `<button>`.
